ops/testing.py

```diff
@@ -900,16 +900,28 @@ def update_config(
     def set_leader(self, is_leader: bool = True) -> None:
         """Set whether this unit is the leader or not.
 
-        If this charm becomes a leader then `leader_elected` will be triggered.
+        If this charm becomes a leader then `leader_elected` will be triggered.  If Harness.begin()
+        has already been called, then the charm's peer relation must be added  *prior* to calling
+        this method (i.e. with Harness.add_relation).
 
         Args:
             is_leader: True/False as to whether this unit is the leader.
         """
-        was_leader = self._backend._is_leader
         self._backend._is_leader = is_leader
+
+        have_peer_relation = False
+        for relid, params in self._backend._relation_app_and_units.items():
+            if params['app'] == self._backend.app_name:
+                have_peer_relation = True
+                break
```

Comment:
I don't understand this particular way of detecting if you have a peer relation. There is a role associated with relations which should be `peer` IIRC.

Reply:
Yeah - it's probably better to cue from the charm meta and peer role.

Reply:
Fixed.